libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt

```diff
@@ -14,9 +14,7 @@ import retrofit2.Callback
 import retrofit2.Response
 
 @MapboxNavigationModule(MapboxNavigationModuleType.OffboardRouter, skipConfiguration = true)
-class MapboxOffboardRouter(private val offboardRoute: NavigationOffboardRoute) : Router {
-
-    constructor(accessToken: String, context: Context) : this(NavigationOffboardRoute.builder(accessToken, context).build())
+class MapboxOffboardRouter(private val accessToken: String, private val context: Context) : Router {
```

Comment:
through `NavigationOffboardRoute` user could setup properties like `roundaboutExits`, `steps` and etc. ptal at [line](https://github.com/mapbox/mapbox-navigation-android/pull/2342/files#diff-83440119e0cd57dea10122af53d120efR11). And if we hide this class and pass default params only how user might change them? 
IMO we should keep this class as a param for `MapboxOffboardRouter`. wdyt? **cc** @LukasPaczos @Guardiola31337

Reply:
Why are those options not in `RouteOptionsNavigation`?

Reply:
good question. I think it's because we decided that this properties are set-upped only once 🤔

Reply:
Huh, as far as I can tell, all of those options are already in the `RouteOptionsNavigation` so they can be overwritten anyway.

If we want to treat https://github.com/mapbox/mapbox-navigation-android/pull/2342/files#diff-83440119e0cd57dea10122af53d120efR12-R23 as smart default, let's set that up in the `RouteOptionsNavigation` themselves. How does that sound?

Reply:
At the moment, most of the options can be set via `RouteOptionsNavigation` but not all of them. We could include them as part of `RouteOptionsNavigation` although we might need to 👀 if it'd make sense as `RouteOptionsNavigation` is going to be shared by offboard and onboard routers. All this comes from keeping parity with the legacy [`NavigationRoute`](https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.kt).

Reply:
yeah,` RouteOptionsNavigation` is common for `Onboard` and `Offboard`. I agree that `RouteOptionsNavigation` might include options from `NavigationRoute`. But some of them couldn't be included because they different for `Offboard` and `Onboard` instances

Reply:
Can we use abstraction in this case? Have base options extended by either offboard and onboard. This would simplify this builder inside of the builder setup.

Reply:
let's merge(comment) this [PR](https://github.com/mapbox/mapbox-navigation-android/pull/2300) to take a look how it's possible